client.go

```diff
@@ -261,10 +285,12 @@ func (c *Client) runForever() {
 			time.Sleep(500 * time.Millisecond)
 		}
 
+		// Tell c.Close() that we have finished shutdown and that it can return.
+		close(c.didStopChan)
```

Comment:
Would it be and idea to expose `isRunning` instead which effectively will say the same thing? Also, if the client is running (that is still in the shutdown process) a new client will not connect. The difference here will be that with this new channel the end user will not be in charge of when to retry after `Stop()` since it's blocking.

Reply:
I'm not sure if I'm following. The client would start again automatically when `.Send()` is called.

If `.Stop()` isn't blocking, doing `client.Stop(); client.Send(...)` could technically have the message added to the `responses` chan while not starting the connections again.

Reply:
Yeah my bad, it just won't create a new connection and trigger `runOnce`, didn't read properly. Would still be nice with a single indicator telling if the client is running (connected) or not but that might be a future improvement.

Reply:
Yea, that would be nice, perhaps a `client.IsConnected()` or something?